include/mqtt/endpoint.hpp

```diff
@@ -684,8 +686,9 @@ class endpoint : public std::enable_shared_from_this<endpoint<Mutex, LockGuard,
      *
      * @param ec error code
      */
-    virtual void on_error(error_code ec) noexcept = 0;
+    virtual void on_error(error_code /*ec*/) noexcept {}
```

Comment:
Instead of making this an empty implementation, i would rather put an empty implementation into client.hpp as an override (but not final)

Reply:
same as https://github.com/redboltz/mqtt_cpp/pull/576#discussion_r364578346